### PR TITLE
[Domains] Consolidate featured suggestion styling

### PR DESCRIPTION
**Pull request by jsnmoon**
Extracted refactoring changes from #24515.

This change consolidates all featured suggestion styles into `/components/domains/featured-domain-suggestions/`. Unnecessary or redundant styles were removed (i.e. specifying `flex` along with `flex-grow`). The interface remains visually identical.

wordpress_com

# Testing instructions
1. Spin up this branch locally.
2. Navigate to `/start/domains`
3. Ensure that the interface renders as expected.

### CODE REVIEW

**Comment by klimeryk** [comment]
Does this need a feature enabled/disabled? I see slightly different UI on staging than on dev (the filters are present on dev, for example).

File: client/components/domains/featured-domain-suggestions/styles/base.scss [hunk 1]
```diff
@@ -85,9 +93,45 @@
...
+			}
+			.featured-domain-suggestions--title-in-12em & {
+				font-size: 1.2em;
+			}
+			.featured-domain-suggestions--title-in-10em & {
+				font-size: 1em;
+			}
+		}
+
+		.featured-domain-suggestions--title-causes-overflow {
```
Comment (delputnam) [inline client/components/domains/featured-domain-suggestions/styles/base.scss]: I think this should be `&.featured-domain-suggestions--title-causes-overflow`.

The way this is implemented prevents these styles from being applied when the title causes an overflow.

wordpress_com

File: client/components/domains/featured-domain-suggestions/styles/base.scss [hunk 2]
```diff
@@ -85,9 +93,45 @@
...
+		.featured-domain-suggestion .domain-suggestion__action.button.is-primary {
+			width: initial;
+		}
+	}
+}
+
+@include breakpoint( '>660px' ) {
+	.featured-domain-suggestions {
+		.domain-registration-suggestion__title {
+			.featured-domain-suggestions--title-in-18em & {
```
Comment (delputnam) [inline client/components/domains/featured-domain-suggestions/styles/base.scss]: I might be missing something, but I don't see where these are still used. If they aren't used, can we remove them?

**Comment by jsnmoon** [comment]
> Does this need a feature enabled/disabled? I see slightly different UI on staging than on dev (the filters are present on dev, for example).

@klimeryk, you can add a query string on development to disable features not yet enabled on production (`?flags=-domains/kracken-ui/filters,-domains/kracken-ui/pagination`). That should get you the same screenshot pictured in the original post.

File: client/components/domains/featured-domain-suggestions/styles/base.scss (hunk 1 above)
Comment (jsnmoon) [reply client/components/domains/featured-domain-suggestions/styles/base.scss]: D'oh! I'll look into this.

File: client/components/domains/featured-domain-suggestions/styles/base.scss (hunk 2 above)
Comment (jsnmoon) [reply client/components/domains/featured-domain-suggestions/styles/base.scss]: Hi Del, this is being used [here](https://github.com/Automattic/wp-calypso/blob/692f4a15ad116f5850739d436f54118a572fedcd/client/components/domains/featured-domain-suggestions/index.jsx#L49) in order to dynamically reduce the font size according to the domain name length.

**Pull request merged**